### ISSUE TITLE
proftpd-1.3.7 expects files in root (/), looks related to ${runstatedir}

### ISSUE THREAD

**Issue by trofi** [proftpd/proftpd#1057]
When updating from `1.3.7_rc4` to `1.3.7` I noticed that a few files like `proftpd.sock` moved from `/var/run/proftpd.sock` to `/proftpd.sock`. It looks related to default-unset:

```
$ git grep -E 'runstatedir' | cat
configure:RUNSTATEDIR=`eval echo $runstatedir`
configure:#define PR_RUN_DIR "`eval echo "${runstatedir}"`"
configure:#define PR_PID_FILE_PATH "`eval echo "${runstatedir}/proftpd.pid"`"
configure.in:RUNSTATEDIR=`eval echo $runstatedir`
configure.in:AC_DEFINE_UNQUOTED(PR_RUN_DIR, "`eval echo "${runstatedir}"`", [Define the run directory])
configure.in:AC_DEFINE_UNQUOTED(PR_PID_FILE_PATH, "`eval echo "${runstatedir}/proftpd.pid"`", [Define the PID file path])
```

It looks like the code now assumes that `runstatedir` should be a non-empty environment variable.

**Comment by trofi** [comment]
If I rerun `autoconf` I see `runstatedir` gets added to known parameters:

```
$ autoconf
$ git diff | cat
```
```diff
--- a/configure
+++ b/configure
@@ -906,6 +906,7 @@ infodir
 docdir
 oldincludedir
 includedir
+runstatedir
 localstatedir
 sharedstatedir
 sysconfdir
@@ -1043,6 +1044,7 @@ datadir='${datarootdir}'
 sysconfdir='${prefix}/etc'
 sharedstatedir='${prefix}/com'
 localstatedir='${prefix}/var'
+runstatedir='${localstatedir}/run'
 includedir='${prefix}/include'
 oldincludedir='/usr/include'
 docdir='${datarootdir}/doc/${PACKAGE}'
@@ -1295,6 +1297,15 @@ do
   | -silent | --silent | --silen | --sile | --sil)
     silent=yes ;;

+  -runstatedir | --runstatedir | --runstatedi | --runstated \
+  | --runstate | --runstat | --runsta | --runst | --runs \
+  | --run | --ru | --r)
+    ac_prev=runstatedir ;;
+  -runstatedir=* | --runstatedir=* | --runstatedi=* | --runstated=* \
+  | --runstate=* | --runstat=* | --runsta=* | --runst=* | --runs=* \
+  | --run=* | --ru=* | --r=*)
+    runstatedir=$ac_optarg ;;
+
   -sbindir | --sbindir | --sbindi | --sbind | --sbin | --sbi | --sb)
     ac_prev=sbindir ;;
   -sbindir=* | --sbindir=* | --sbindi=* | --sbind=* | --sbin=* \
@@ -1432,7 +1443,7 @@ fi
 for ac_var in  exec_prefix prefix bindir sbindir libexecdir datarootdir \
                datadir sysconfdir sharedstatedir localstatedir includedir \
                oldincludedir docdir infodir htmldir dvidir pdfdir psdir \
-               libdir localedir mandir
+               libdir localedir mandir runstatedir
 do
   eval ac_val=\$$ac_var
   # Remove trailing slashes.
@@ -1585,6 +1596,7 @@ Fine tuning of the installation directories:
   --sysconfdir=DIR        read-only single-machine data [PREFIX/etc]
   --sharedstatedir=DIR    modifiable architecture-independent data [PREFIX/com]
   --localstatedir=DIR     modifiable single-machine data [PREFIX/var]
+  --runstatedir=DIR       modifiable per-process data [LOCALSTATEDIR/run]
...
```

Could it be it's just missing ./configure hunk?

**Comment by Castaglia** [comment]
See Issue #1055

**Comment by trofi** [comment]
Ah, thank you!

**Comment by Castaglia** [comment]
Yeah, this is subtle, and depends on the _version_ of Autoconf being used.  The last released Autoconf version, 2.69, does _not_ have support for `--runstatedir` -- but many packagers are using the unreleased Autoconf 2.70, which _does_.   Tricksy.
